src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T value) {
+            this.value = value;
+        }
+    }
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            tail = head = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index > size || index < 0) {
```

Comment:
```suggestion
        if (index < 0 || index > size) {
```
a minor issue but IMO it's more readable when conditions are in a linear direction, from small number to big numbers